hevm.cabal

```diff
@@ -108,17 +111,9 @@ library
     Paths_hevm
   autogen-modules:
     Paths_hevm
-  if impl(ghc >= 9.4) && !os(darwin)
-    -- darwin is skipped because it produces this error when building
-    -- > ghc: loadArchive: Neither an archive, nor a fat archive: `/nix/store/l3lkdfm7sg1wwc850451cikqds766h15-clang-wrapper-11.1.0/bin/clang++'
+  if os(linux)
     build-depends: system-cxx-std-lib
-  elif !os(darwin)
     extra-libraries: stdc++
```

Comment:
I think we may be able to simplify this now and just have `build-depends: system-cxx-std-lib` for all platforms, plus the special darwin flags just in one place.

Reply:
@elopez , do you want to try it?
There are already 4 people contributing to this PR, why not make it 5 😄

Reply:
Your commits are included. First, there was some problem in CI, but after restarting a job, everything is green now!